epochX/cudacpp/CODEGEN/PLUGIN/CUDACPP_SA_OUTPUT/model_handling.py

```diff
@@ -823,15 +823,30 @@ def super_generate_parameters_class_files(self):
         """Create the content of the Parameters_model.h and .cc files"""
         # First of all, identify which extra independent parameters must be made available through CPU static and GPU constant memory in BSM models
         # because they are used in the event by event calculation of alphaS-dependent couplings
-        # WARNING! This is only implemented and has only been tested so far for real parameters (complex parameters need twice the storage)
         bsmparam_indep_real_used = []
-        if self.model_name[:2] != 'sm' :
-            for param in self.params_indep:
-                if param.type == 'real':
-                    for coup in self.coups_dep.values():                
-                        if param.name in coup.expr:
-                            bsmparam_indep_real_used.append( param.name )
-            bsmparam_indep_real_used = set( bsmparam_indep_real_used ) 
+        bsmparam_indep_complex_used = []
+        for param in self.params_indep: # NB this is now done also for 'sm' processes (no check on model name, see PR #824)
+            if param.name == 'mdl_complexi' : continue
+            if param.name == 'aS' : continue
```

Comment:
Thanks @oliviermattelaer . As I imagined, we already have since a long time a treatment of 'ZERO':
https://github.com/madgraph5/madgraph4gpu/blob/df29acb070abeb219b65facd8b2b451b27f36e5e/epochX/cudacpp/CODEGEN/PLUGIN/CUDACPP_SA_OUTPUT/model_handling.py#L1710

So nothing changed in this MR about this. This check is in another part of the code, and I replace 'ZERO' with a hardcoded '0'. Let me know if you see an issue with this.